### ISSUE TITLE
Gaussian Mixture Model

### ISSUE THREAD

**Issue by chananshgong** [jmschrei/pomegranate#223]
How do I fit data to a Gaussian Mixture Model?
Do you have a preliminary fit process like in scikit-learn gmm package?

**Comment by jkleckner** [comment]
Most of you questions should be answered in the examples.  Have you looked at for instance http://pomegranate.readthedocs.io/en/latest/GeneralMixtureModel.html#fitting ?

Also try out the notebooks which are easy to enable and play around with them.

**Comment by jmschrei** [comment]
I use kmeans like sklearn does to get an initial estimate, and then use those values to initiate EM. I only use one iteration of kmeans though.  You may also be interested in this https://github.com/jmschrei/pomegranate/blob/master/benchmarks/pomegranate_vs_sklearn_gmm.ipynb where I directly compare the two packages.

**Comment by chanansh** [comment]
Thanks @jmschrei  for the benchmark notebook. 
I am worried by the quote at the end of the comparison associated with last accuracy plot:

> "It seems as though both fitting and prediction using pomegranate scale less well than sklearn to more dimensions, but that it is still faster."

Does that mean pomegranate does a bad job at higher dimensions? (chance level?). I need at least 20 dimensions.

**Comment by jmschrei** [comment]
Hi @chanansh 

Sorry for taking a bit to get back to you. There was a bug in that cell. In particular, while sklearn's GMM can be refit to different dimensional datasets, pomegranate's cannot. This was the reason that it performed really poorly-- it assumed that all of the higher dimensional data was 2D, because it was originally fit to 2D data. I made a modification to the cell to adjust for it and reuploaded the notebook. Check it out and let me know if you have any other questions.  It shouldn't perform much worse than sklearn. What I meant by that line is that the speed improvement diminishes as a function of dimensions, but that it seems to be faster. In particular, if you look at 20 dimensional data in the example you see that it is around twice as fast as sklearn at fitting the model.

Also keep in mind that pomegranate supports parallelized training of all models, including GMM, in a data-parallel manner. You can use the `fit` function defined here: https://github.com/jmschrei/pomegranate/blob/master/pomegranate/parallel.pyx#L328 to get faster training. Let me know how it works for you!
